VersionControlUI/src/items/VDiffComparisonPair.cpp

```diff
@@ -55,18 +58,67 @@ void VDiffComparisonPair::initializeForms()
 			return v->node()->newVersionObjectPath();},
 			[](I* v) {return &v->style()->newVersionObjectPath();});
 
+	auto componentType = item<Visualization::VText>(&I::componentType_, [](I* v) {
+			return v->node()->componentType();},
+			[](I* v) {return &v->style()->componentType();});
+
+	auto infoGrid = (new Visualization::GridLayoutFormElement{})
+			->setHorizontalSpacing(50)
+			->setLeftMargin(10)
+			->setRightMargin(10)
+			->setHorizontalAlignment(Visualization::LayoutStyle::Alignment::Center)
+			->setNoBoundaryCursors([](Item*){return true;})->setNoInnerCursors([](Item*){return true;})
+			->setColumnStretchFactor(0, 1)
+			->setColumnStretchFactor(1, 1)
+			->put(0, 0, oldVersionObjectPath)
+			->put(1, 0, newVersionObjectPath)
+			->put(0, 1, componentType);
 	auto container = (new Visualization::GridLayoutFormElement{})
 				->setHorizontalSpacing(30)
 				->setVerticalSpacing(15)
 				->setHorizontalAlignment(Visualization::LayoutStyle::Alignment::Center)
 				->setNoBoundaryCursors([](Item*){return true;})->setNoInnerCursors([](Item*){return true;})
-				->put(0, 0, oldVersionObjectPath)
-				->put(1, 0, newVersionObjectPath)
+				->setLeftMargin(10)
+				->setRightMargin(10)
+				->setColumnStretchFactor(0, 1)
+				->setColumnStretchFactor(1, 1)
+				->setColumnHorizontalAlignment(0, Visualization::LayoutStyle::Alignment::Right)
+				->setColumnHorizontalAlignment(1, Visualization::LayoutStyle::Alignment::Left)
+				->put(0, 0, infoGrid)
+				->setCellSpanning(2, 1)
 				->put(0, 1, oldVersion)
 				->put(1, 1, newVersion);
 
 	addForm(container);
 
 }
 
+bool VDiffComparisonPair::isSensitiveToScale() const
+{
+	return true;
+}
+
+void VDiffComparisonPair::determineChildren()
+{
+	Super::determineChildren();
+	qreal factor = Visualization::VisualizationManager::instance().mainScene()->mainViewScalingFactor();
```

Comment:
To make what's going on here a little more isolated, please put all the code after the super call in a separate private method.